src/java/org/apache/cassandra/index/sai/disk/SSTableIndex.java

```diff
@@ -238,23 +247,25 @@ public boolean equals(Object o)
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         SSTableIndex other = (SSTableIndex)o;
-        return Objects.equal(sstableContext, other.sstableContext) && Objects.equal(indexContext, other.indexContext);
+        return Objects.equal(sstableContext, other.sstableContext) &&
+               Objects.equal(indexTermType, other.indexTermType) &&
+               Objects.equal(indexIdentifier, other.indexIdentifier);
```

Comment:
Does `IndexIdentifier` need to implement `equals()` and `hashCode()`?

Reply:
Yes, it does. I've added them.